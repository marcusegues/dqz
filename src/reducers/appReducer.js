/* eslint-disable prefer-destructuring */
// @flow
import Immutable from 'immutable';
import {
  getInitialState,
  mainCategories,
  EmptyMainCategories,
} from '../types/reducers/appReducer';
import type {
  State,
  MainCategory,
  MainCategories,
  Settings,
} from '../types/reducers/appReducer';
import type { Action } from '../types/actions';
import type {
  Amounts,
  Basket,
  Category,
  People,
} from '../model/types/basketPeopleAmountsTypes';
import * as modelApi from '../model/configurationApi';
import type { Currency, CurrencyObject } from '../model/currencies';
import type { PaymentData } from '../types/generalTypes';

export const appState = (
  state: State = getInitialState(),
  action: Action
): State => {
  switch (action.type) {
    case 'UPDATE_CURRENCIES': {
      const currencyObject: CurrencyObject = action.currencyObject;
      const validCurrencies: boolean = action.validCurrencies;
      const currencyDate: Date = action.currencyDate;

      const s1 = state.set('currencyObject', currencyObject);
      const s2 = s1.set('currencyDate', currencyDate);
      return s2.set('validCurrencies', validCurrencies);
    }

    case 'BASKET_ADD_QUANTITY': {
      const category: Category = action.category;
      const date: string = action.date;
      const basket: Basket = state.get('basket');
      return state.setIn(
        ['basket'],
        modelApi.addQuantity(
          basket,
          category,
          action.quantity,
          date.length > 0 ? date : undefined
        )
      );
    }
    case 'SET_BASKET': {
      const basket: Basket = action.basket;
      return state.set('basket', basket);
    }

    case 'ADULTS_CHANGE_QUANTITY': {
      const people: People = state.get('people');
      const adults: number = people.get('adults');
      const quantity: number = adults + action.quantityChange;
      return state.set('people', modelApi.setAdultPeople(people, quantity));
    }
    case 'MINORS_CHANGE_QUANTITY': {
      const people: People = state.get('people');
      const minors: number = people.get('minors');
      const quantity: number = minors + action.quantityChange;
      return state.set('people', modelApi.setMinorPeople(people, quantity));
    }
    case 'ADULTS_SET_QUANTITY': {
      const people: People = state.get('people');
      const quantity: number = action.quantity;
      return state.set('people', modelApi.setAdultPeople(people, quantity));
    }
    case 'MINORS_SET_QUANTITY': {
      const people: People = state.get('people');
      const quantity: number = action.quantity;
      return state.set('people', modelApi.setMinorPeople(people, quantity));
    }
    case 'SET_PEOPLE': {
      return state.set('people', action.people);
    }

    case 'SET_AMOUNTS': {
      const amounts: Amounts = action.amounts;
      return state.set('amounts', amounts);
    }
    case 'ADD_AMOUNT': {
      const currency: Currency = action.currency;
      const amounts: Amounts = state.get('amounts');
      const amount: number = action.amount;
      return state.set(
        'amounts',
        modelApi.addAmount(amounts, currency, amount)
      );
    }
    case 'ADD_LARGE_AMOUNT': {
      const currency: Currency = action.currency;
      const largeAmount: number = action.largeAmount;
      const amounts: Amounts = state.get('amounts');
      return state.set(
        'amounts',
        modelApi.addLargeAmount(amounts, currency, largeAmount)
      );
    }
    case 'RESET_LARGE_AMOUNTS': {
      const amounts: Amounts = state.get('amounts');
      const currency: Currency = action.currency;
      return state.set(
        'amounts',
        modelApi.resetLargeAmounts(amounts, currency)
      );
    }
    case 'RESET_AMOUNTS': {
      const amounts: Amounts = state.get('amounts');
      const currency: Currency = action.currency;
      return state.set('amounts', modelApi.resetAmounts(amounts, currency));
    }

    case 'RESET_DECLARATION': {
      const peopleReset = state.set('people', modelApi.initPeople);
      const basketReset = peopleReset.set('basket', modelApi.emptyBasket);
      const mainCategoriesReset = basketReset.setIn(
        ['settings', 'mainCategories'],
        Immutable.Set()
      );
      return mainCategoriesReset.set('amounts', modelApi.initAmounts);
    }

    case 'ADD_MAIN_CATEGORY': {
      const mainCategory: MainCategory = action.mainCategory;
      const mainCategoriesAnswer = state.getIn(
        ['settings', 'mainCategories'],
        EmptyMainCategories
      );
      return state.setIn(
        ['settings', 'mainCategories'],
        mainCategoriesAnswer.add(mainCategory)
      );
    }
    case 'REMOVE_MAIN_CATEGORY': {
      const mainCategory: MainCategory = action.mainCategory;
      const mainCategoriesAnswer = state.getIn(
        ['settings', 'mainCategories'],
        EmptyMainCategories
      );
      return state.setIn(
        ['settings', 'mainCategories'],
        mainCategoriesAnswer.delete(mainCategory)
      );
    }
    case 'SET_MAIN_CATEGORIES': {
      const mainCategoriesAnswer: MainCategories = action.mainCategories;
      return state.setIn(['settings', 'mainCategories'], mainCategoriesAnswer);
    }

    case 'SET_PAYMENT_DATA': {
      const paymentData: PaymentData = action.paymentData;
      return state.set('paymentData', paymentData);
    }

    case 'SET_RECEIPT_ID': {
      const receiptId: string = action.receiptId;
      return state.set('receiptId', receiptId);
    }

    case 'SET_RECEIPT_ENTRY_TIME': {
      const receiptEntryTime: string = action.receiptEntryTime;
      return state.set('receiptEntryTime', receiptEntryTime);
    }

    default: {
      return state;
    }
  }
};

export const getBasket = (state: State): Basket => state.get('basket');

export const getPeople = (state: State): People => state.get('people');

export const getAmounts = (state: State): Amounts => state.get('amounts');

export const getMainCategories = (state: State): MainCategories =>
  state.getIn(['settings', 'mainCategories'], mainCategories);

export const getSettings = (state: State): Settings => state.get('settings');

export const getCurrenciesObject = (state: State): CurrencyObject =>
  state.get('currencyObject');

export const getCurrencyState = (state: State): boolean =>
  state.get('validCurrencies');

export const getCurrencyDate = (state: State): Date =>
  state.get('currencyDate');

export const getPaymentData = (state: State): PaymentData =>
  state.get('paymentData');

export const getReceiptId = (state: State): string => state.get('receiptId');

export const getReceiptEntryTime = (state: State): string =>
  state.get('receiptEntryTime');
