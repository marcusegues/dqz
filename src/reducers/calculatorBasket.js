import { emptyBasket } from '../constants/basket';

const calculatorBasket = (state = emptyBasket, action) => {
  switch (action.type) {
    case 'CHANGE_QUANTITY_CALCULATOR_BASKET_ITEM': {
      return {
        ...state,
        [action.itemName]: state[action.itemName] + action.quantityChange,
      };
    }
    default: {
      return state;
    }
  }
};

export default calculatorBasket;
