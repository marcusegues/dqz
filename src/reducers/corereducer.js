const corereducer = (state = {}, action) => {
  switch (action.type) {
    default: {
      return state;
    }
  }
};

export default corereducer;
