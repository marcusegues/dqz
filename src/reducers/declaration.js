// @flow
import {
  getInitialState,
  MainCategories,
  InitList,
} from '../types/reducers/declaration';
import type {
  State,
  MainCategory,
  MainCategoriesType,
  OverAllowanceType,
  LargeAmountPresentType,
  SettingsType,
  CurrentQuestionType,
} from '../types/reducers/declaration';
import type { Action } from '../types/actions';
import type {
  Basket,
  Category,
  People,
} from '../model/types/basketPeopleTypes';
import * as fromModelApi from '../model/configurationApi';

const declaration = (
  state: State = getInitialState(),
  action: Action
): State => {
  switch (action.type) {
    case 'DECLARATION_BASKET_CHANGE_QUANTITY': {
      // eslint-disable-next-line prefer-destructuring
      const category: Category = action.category;
      const basket: Basket = state.get('basket');

      return state.setIn(
        ['basket'],
        fromModelApi.setQuantity(
          basket,
          category,
          fromModelApi.getQuantity(basket, category) + action.quantityChange
        )
      );
    }
    case 'DECLARATION_BASKET_ADD_AMOUNT': {
      // eslint-disable-next-line prefer-destructuring
      const category: Category = action.category;
      const basket: Basket = state.get('basket');
      // eslint-disable-next-line prefer-destructuring
      const amount: number = action.amount;
      return state.set(
        'basket',
        fromModelApi.addAmount(basket, category, amount)
      );
    }
    case 'DECLARATION_BASKET_ADD_LARGE_AMOUNT': {
      // eslint-disable-next-line prefer-destructuring
      const category: Category = action.category;
      // eslint-disable-next-line prefer-destructuring
      const largeAmount: number = action.largeAmount;
      const basket: Basket = state.get('basket');
      return state.set(
        'basket',
        fromModelApi.addLargeAmount(basket, category, largeAmount)
      );
    }
    case 'DECLARATION_ADULTS_CHANGE_QUANTITY': {
      const people: People = state.get('people');
      const adults: number = people.get('adults');
      const quantity: number = adults + action.quantityChange;
      return state.set('people', fromModelApi.setAdultPeople(people, quantity));
    }
    case 'DECLARATION_MINORS_CHANGE_QUANTITY': {
      const people: People = state.get('people');
      const minors: number = people.get('minors');
      const quantity: number = minors + action.quantityChange;
      return state.set('people', fromModelApi.setMinorPeople(people, quantity));
    }
    case 'DECLARATION_ADULTS_SET_QUANTITY': {
      const people: People = state.get('people');
      // eslint-disable-next-line prefer-destructuring
      const quantity: number = action.quantity;
      return state.set('people', fromModelApi.setAdultPeople(people, quantity));
    }
    case 'DECLARATION_MINORS_SET_QUANTITY': {
      const people: People = state.get('people');
      // eslint-disable-next-line prefer-destructuring
      const quantity: number = action.quantity;
      return state.set('people', fromModelApi.setMinorPeople(people, quantity));
    }
    case 'DECLARATION_SET_PEOPLE': {
      const people: People = state.get('people');
      // eslint-disable-next-line prefer-destructuring
      const adults: number = action.adults;
      // eslint-disable-next-line prefer-destructuring
      const minors: number = action.minors;
      return state.set(
        'people',
        fromModelApi.setPeople(people, adults, minors)
      );
    }
    case 'DECLARATION_SET_OVER_ALLOWANCE_TRUE': {
      return state.setIn(['settings', 'overAllowance'], true);
    }
    case 'DECLARATION_SET_OVER_ALLOWANCE_FALSE': {
      return state.setIn(['settings', 'overAllowance'], false);
    }
    case 'DECLARATION_SET_LARGE_AMOUNT_PRESENT_TRUE': {
      return state.setIn(['settings', 'largeAmountPresent'], true);
    }
    case 'DECLARATION_SET_LARGE_AMOUNT_PRESENT_FALSE': {
      return state.setIn(['settings', 'largeAmountPresent'], false);
    }
    case 'DECLARATION_SET_LARGE_AMOUNT_PRESENT_NOT_ANSWERED': {
      return state.setIn(['settings', 'largeAmountPresent'], 'notAnswered');
    }
    case 'DECLARATION_RESET_LARGE_AMOUNTS': {
      const basket: Basket = state.get('basket');
      // eslint-disable-next-line prefer-destructuring
      const category: Category = action.category;
      return state.set(
        'basket',
        fromModelApi.resetLargeAmounts(basket, category)
      );
    }
    case 'DECLARATION_SET_LARGE_AMOUNTS_ENTERED_TRUE': {
      return state.setIn(['settings', 'largeAmountsEntered'], true);
    }
    case 'DECLARATION_SET_LARGE_AMOUNTS_ENTERED_NOT_ANSWERED': {
      debugger;
      return state.setIn(['settings', 'largeAmountsEntered'], 'notAnswered');
    }
    case 'DECLARATION_SET_OVER_ALLOWANCE_NOT_ANSWERED': {
      return state.setIn(['settings', 'overAllowance'], 'notAnswered');
    }
    case 'DECLARATION_RESET_AMOUNTS': {
      const basket: Basket = state.get('basket');
      // eslint-disable-next-line prefer-destructuring
      const category: Category = action.category;
      return state.set('basket', fromModelApi.resetAmounts(basket, category));
    }
    case 'DECLARATION_ADD_MAIN_CATEGORY': {
      // eslint-disable-next-line prefer-destructuring
      const mainCategory: MainCategory = action.mainCategory; // why can't I omit the declaration and pass directly into add?
      return state.updateIn(['settings', 'mainCategories'], mainCategories =>
        mainCategories.add(mainCategory)
      );
    }
    case 'DECLARATION_REMOVE_MAIN_CATEGORY': {
      // eslint-disable-next-line prefer-destructuring
      const mainCategory: MainCategory = action.mainCategory;
      return state.updateIn(['settings', 'mainCategories'], mainCategories =>
        mainCategories.delete(mainCategory)
      );
    }
    case 'DECLARATION_SET_MAIN_CATEGORIES': {
      // eslint-disable-next-line prefer-destructuring
      const mainCategories: MainCategoriesType = action.mainCategories;
      return state.setIn(['settings', 'mainCategories'], mainCategories);
    }
    case 'DECLARATION_SET_CURRENT_QUESTION': {
      // eslint-disable-next-line prefer-destructuring
      const currentQuestion: CurrentQuestionType = action.currentQuestion;
      return state.setIn(['settings', 'currentQuestion'], currentQuestion);
    }
    case 'DECLARATION_ADD_TO_INIT_LIST': {
      const nextQuestion: CurrentQuestionType = action.nextQuestion;
      const newState = state.updateIn(['settings', 'initList'], list =>
        list.push(nextQuestion)
      );
      return newState;
    }
    case 'DECLARATION_SET_INIT_FALSE': {
      const newState = state.setIn(['settings', 'init'], false);
      return newState.setIn(['settings', 'initList'], InitList);
    }
    default: {
      return state;
    }
  }
};

export default declaration;

export const getDeclarationBasket = (state: State): Basket =>
  state.get('basket');

export const getDeclarationPeople = (state: State): People =>
  state.get('people');

export const getOverAllowance = (state: State): OverAllowanceType =>
  state.getIn(['settings', 'overAllowance'], 'notAnswered');

export const getLargeAmountPresent = (state: State): LargeAmountPresentType =>
  state.getIn(['settings', 'largeAmountPresent'], 'notAnswered');

export const getDeclarationMainCategories = (
  state: State
): MainCategoriesType =>
  state.getIn(['settings', 'mainCategories'], MainCategories);

export const getDeclarationCurrentQuestion = (
  state: State
): CurrentQuestionType =>
  state.getIn(['settings', 'currentQuestion'], 'finished');

export const getDeclarationSettings = (state: State): SettingsType =>
  state.get('settings');

export const getDeclarationInit = (state: State): boolean =>
  state.getIn(['settings', 'init'], true);

export const getDeclarationInitList = (state: State): boolean =>
  state.getIn(['settings', 'initList'], true);
