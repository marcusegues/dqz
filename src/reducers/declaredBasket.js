import { emptyBasket } from '../constants/basket';

const declaredBasket = (state = emptyBasket, action) => {
  switch (action.type) {
    case 'INCREMENT_DECLARED_BASKET_ITEM': {
      return {
        ...state,
        [action.itemName]: state[action.itemName] + action.increment,
      };
    }
    default: {
      return state;
    }
  }
};

export default declaredBasket;
