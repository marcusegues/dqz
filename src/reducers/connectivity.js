import type { State } from '../types/reducers/appReducer';
import { getInitialState } from '../types/reducers/connectivityReducer';
import type { ConnectivityState } from '../types/reducers/connectivityReducer';
import type { ConnectivityAction } from '../types/actions/connectivity';

export const connectivity = (
  state: ConnectivityState = getInitialState(),
  action: ConnectivityAction
): State => {
  switch (action.type) {
    case 'SET_CONNECTIVITY': {
      return action.connectionInfo;
    }
    default: {
      return state;
    }
  }
};
