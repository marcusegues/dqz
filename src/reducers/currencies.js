/* eslint-disable prefer-destructuring */
// @flow
import type { CurrenciesState } from '../types/reducers/currencies';
import { getInitialCurrenciesState } from '../types/reducers/currencies';
import type { CurrenciesAction } from '../types/actions/currencies';
import type { CurrencyObject } from '../model/currencies';

export const currencies = (
  state: CurrenciesState = getInitialCurrenciesState(),
  action: CurrenciesAction
): CurrenciesState => {
  switch (action.type) {
    case 'UPDATE_CURRENCIES': {
      const currencyObject: CurrencyObject = action.currencyObject;
      const validCurrencies: boolean = action.validCurrencies;
      const currencyDate: Date = action.currencyDate;

      const s1 = state.set('currencyObject', currencyObject);
      const s2 = s1.set('currencyDate', currencyDate);
      return s2.set('validCurrencies', validCurrencies);
    }
    default: {
      return state;
    }
  }
};

export const getCurrenciesObject = (state: CurrenciesState): CurrencyObject =>
  state.get('currencyObject');

export const getCurrencyState = (state: CurrenciesState): boolean =>
  state.get('validCurrencies');

export const getCurrencyDate = (state: CurrenciesState): Date =>
  state.get('currencyDate');
