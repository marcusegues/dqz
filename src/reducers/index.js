// @flow
import { combineReducers } from 'redux';
import * as appApi from './appReducer';
import * as connectivityApi from './connectivity';
import { connectivity } from './connectivity';
import { formatDate } from '../model/utils';
import type {
  Amounts,
  Basket,
  People,
} from '../model/types/basketPeopleAmountsTypes';
import type {
  MainCategories,
  Settings,
  State,
} from '../types/reducers/appReducer';
import type { CurrencyObject } from '../model/currencies';
import { calculateVat } from '../model/vatCalculations';
import { calculateDuty } from '../model/dutyCalculations';
import type { DutyReport, VatReport } from '../model/types/calculationTypes';
import type { PaymentData } from '../types/generalTypes';
import type { ConnectivityType } from '../types/connectivity';
import type { AppState } from '../types/reducers';

/**
 * Combines reducers (currently just one)
 * @type {Reducer<any>}
 */
export const root: AppState = combineReducers({
  appState: appApi.appState,
  connectivity,
});

/**
 * Gets basket in application state
 * @param state
 * @returns {Basket}
 */
export const getBasket = (state: AppState): Basket =>
  appApi.getBasket(state.appState);

/**
 * Gets people in application state
 * @param state
 * @returns {People}
 */
export const getPeople = (state: AppState): People =>
  appApi.getPeople(state.appState);

/**
 * Gets amounts in application state
 * @param state
 * @returns {Amounts}
 */
export const getAmounts = (state: AppState): Amounts =>
  appApi.getAmounts(state.appState);

/**
 * Gets main categories in application state
 * @param state
 * @returns {MainCategories}
 */
export const getMainCategories = (state: AppState): MainCategories =>
  appApi.getMainCategories(state.appState);

/**
 * Gets settings in application state
 * @param state
 * @returns {Settings}
 */
export const getSettings = (state: AppState): Settings =>
  appApi.getSettings(state.appState);

/**
 * Gets currencies in application state
 * @param state
 * @returns {CurrencyObject}
 */
export const getCurrencies = (state: AppState): CurrencyObject =>
  appApi.getCurrenciesObject(state.appState);

/**
 * Gets currency state in application state
 * @param state
 * @returns {boolean}
 */
export const getCurrencyState = (state: AppState): boolean =>
  appApi.getCurrencyState(state.appState);

/**
 * Gets the formatted currency date in application state
 * @param state
 * @returns {string}
 */
export const getFormattedCurrencyDate = (state: AppState): string =>
  formatDate(appApi.getCurrencyDate(state.appState));

/**
 * Gets the total vat in application state
 * @param state
 */
export const getTotalVat = (state: AppState): number =>
  calculateVat(
    state.appState.amounts,
    state.appState.people,
    state.appState.currencyObject
  ).get('totalVat', 0);

/**
 * Gets the total duty in application state
 * @param state
 */
export const getTotalDuty = (state: AppState): number =>
  calculateDuty(state.appState.basket, state.appState.people).get(
    'totalDuty',
    0
  );

/**
 * Gets the total fees (== vat + duty) in application state
 * @param state
 * @returns {number}
 */
export const getTotalFees = (state: AppState): number =>
  getTotalVat(state) + getTotalDuty(state);

/**
 * Gets the duty report in application state
 * @param state
 * @returns {DutyReport}
 */
export const getDutyReport = (state: AppState): DutyReport =>
  calculateDuty(state.appState.basket, state.appState.people);

/**
 * Gets the vat report in application state
 * @param state
 * @returns {VatReport}
 */
export const getVatReport = (state: AppState): VatReport =>
  calculateVat(
    state.appState.amounts,
    state.appState.people,
    state.appState.currencyObject
  );

/**
 * Gets current payment data
 * @param state
 * @returns {PaymentData}
 */
export const getPaymentData = (state: AppState): PaymentData =>
  appApi.getPaymentData(state.appState);

/**
 * Gets the current receipt id (emptystring if not active)
 * @param state
 * @returns {string}
 */
export const getReceiptId = (state: AppState): string =>
  appApi.getReceiptId(state.appState);

/**
 * Gets the current receipt entry time (empty string if not set yet)
 * @param state
 * @returns {string}
 */
export const getReceiptEntryTime = (state: AppState): string =>
  appApi.getReceiptEntryTime(state.appState);

/**
 * Gets the current receipt id (emptystring if not active)
 * @param state
 * @returns {string}
 */
export const getConnectivity = (state: AppState): ConnectivityType =>
  connectivityApi.getConnectivity(state.connectivity);
