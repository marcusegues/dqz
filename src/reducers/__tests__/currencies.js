// @flow
import { currencyExample } from '../../model/currencies';
import { getInitialCurrenciesState } from '../../types/reducers/currencies';
import type { CurrenciesState } from '../../types/reducers/currencies';
import {
  currencies,
  getCurrenciesObject,
  getCurrencyDate,
  getCurrencyState,
} from '../currencies';

const initCurrenciesState: CurrenciesState = getInitialCurrenciesState();

describe('Currencies...', () => {
  test('UPDATE_CURRENCIES', () => {
    expect(
      currencies(initCurrenciesState, {
        type: 'UPDATE_CURRENCIES',
        currencyObject: currencyExample,
        validCurrencies: true,
        currencyDate: new Date('2000/01/01'),
        // $FlowFixMe - overriding the date
      }).set('currencyDate', '')
    ).toMatchSnapshot();
  });
});

describe('API', () => {
  test('getCurrenciesObject', () => {
    expect(getCurrenciesObject(initCurrenciesState)).toBe(
      initCurrenciesState.get('currencyObject')
    );
  });
  test('getCurrencyState', () => {
    expect(getCurrencyState(initCurrenciesState)).toBe(
      initCurrenciesState.get('validCurrencies')
    );
  });
  test('getCurrencyDate', () => {
    expect(getCurrencyDate(initCurrenciesState)).toBe(
      initCurrenciesState.get('currencyDate')
    );
  });
});
