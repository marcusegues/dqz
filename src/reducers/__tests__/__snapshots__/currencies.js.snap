// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies... UPDATE_CURRENCIES 1`] = `
Immutable.Record {
  "currencyObject": Object {
    "AED": 0.2649167,
    "ALL": 0.008944,
    "ARS": 0.05195,
    "AUD": 0.77355,
    "AZN": 0.58259,
    "BAM": 0.6101806,
    "BDT": 0.0117518,
    "BGN": 0.6102572,
    "BHD": 2.59435,
    "BRL": 0.3026607,
    "CAD": 0.78292,
    "CHF": 1,
    "CLP": 0.00161429,
    "CNY": 0.1513162,
    "COP": 0.000340239,
    "CRC": 0.00171896,
    "CZK": 0.0468083,
    "DKK": 0.1607132,
    "DOP": 0.0200812,
    "EGP": 0.0549555,
    "ETB": 0.0357578,
    "EUR": 1.17796,
    "GBP": 1.34254,
    "GTQ": 0.1325793,
    "HKD": 0.124335,
    "HNL": 0.0412188,
    "HRK": 0.160991,
    "HUF": 0.00386791,
    "IDR": 0.000073038,
    "ILS": 0.28701,
    "INR": 0.0152953,
    "ISK": 0.009519,
    "JPY": 0.0088304,
    "KES": 0.0094505,
    "KHR": 0.000240016,
    "KRW": 0.0009148,
    "KWD": 3.24657,
    "KYD": 1.1865,
    "KZT": 0.0029631,
    "LBP": 0.00064433,
    "LKR": 0.00632967,
    "LYD": 0.72607,
    "MAD": 0.10617,
    "MUR": 0.0307404,
    "MXN": 0.05169,
    "MYR": 0.246281,
    "NGN": 0.0027139,
    "NOK": 0.1234765,
    "NZD": 0.71013,
    "OMR": 2.53249,
    "PAB": 0.97293,
    "PEN": 0.3030944,
    "PHP": 0.0193061,
    "PKR": 0.0087885,
    "PLN": 0.286039,
    "QAR": 0.2672893,
    "RON": 0.2576692,
    "RSD": 0.0101189,
    "RUB": 0.017273,
    "SAR": 0.259428,
    "SEK": 0.1219009,
    "SGD": 0.73629,
    "THB": 0.0304708,
    "TND": 0.3989229,
    "TRY": 0.2564737,
    "TWD": 0.0329384,
    "TZS": 0.00043415,
    "UAH": 0.0343762,
    "USD": 0.99205,
    "UYU": 0.034126,
    "VEF": 0.0964561,
    "VND": 0.000042925,
    "ZAR": 0.07911,
  },
  "validCurrencies": true,
  "currencyDate": "",
}
`;
